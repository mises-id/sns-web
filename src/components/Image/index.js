/*
 * @Author: lmk
 * @Date: 2021-07-14 21:52:16
 * @LastEditTime: 2022-01-21 21:46:34
 * @LastEditors: lmk
 * @Description:
 */
import React, { useEffect, useState } from "react";
import head from "@/images/head.png";
import image from "@/images/image.png";
/**
 * @description: img
 * @param {*} source src for image
 * @param {*} size image layout
 * @param {*} shape image shape
 * @param {*} alt img type
 * @return {*} element
 */
const Image = ({
  source = head,
  size = "lg",
  shape = "circle",
  alt = "avatar",
  onClick,
  borderRadius = "1px",
  height,
}) => {
  //shape:circle square
  const normal = alt === "avatar" ? head : image;
  const [src, setsrc] = useState(normal);
  borderRadius = shape === "circle" ? "50%" : borderRadius;
  useEffect(() => {
    setsrc(source || head);
  }, [source]);
  const imgSize =
    typeof size === "string"
      ? {
          md: "60px",
          lg: "40px",
          sm: "28px",
          xs: "15px",
        }[size] || size
      : `${size}px`;
  const error = (e) => setsrc(normal);
  return (
    src && (
      <img
        onError={error.bind(this)}
        src={src}
        style={{
          height: height || imgSize,
          width: imgSize,
          objectFit: "cover",
          borderRadius,
        }}
        className={`border-${shape}`}
        alt={alt}
        onClick={onClick}
      />
    )
  );
};
export default Image;
