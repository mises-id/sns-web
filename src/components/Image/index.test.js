/*
 * @Author: lmk
 * @Date: 2021-07-15 10:12:21
 * @LastEditTime: 2021-07-15 10:42:47
 * @LastEditors: lmk
 * @Description: 
 */
import React from 'react';
import {shallow} from 'enzyme';
import Image from '.';
describe('Image', () => {
  it('should render correctly', () => {
    const component = shallow(<Image />)
    expect(component).toMatchSnapshot();
  });
});