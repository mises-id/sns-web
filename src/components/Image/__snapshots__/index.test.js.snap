// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should render correctly 1`] = `
<img
  alt="avatar"
  className="border-circle"
  onError={[Function]}
  src="head.png"
  style={
    Object {
      "height": "40px",
      "objectFit": "cover",
      "width": "40px",
    }
  }
/>
`;
